app/apphandlers/processorAppHandler.go

```diff
@@ -179,15 +178,24 @@ func (processor *ProcessorApp) StartRudderCore(ctx context.Context, options *app
 			})
 		}
 	}
-
-	var modeProvider *state.StaticProvider
-	// FIXME: hacky way to determine servermode
-	if enableProcessor && enableRouter {
-		modeProvider = state.NewStaticProvider(servermode.NormalMode)
-	} else {
-		modeProvider = state.NewStaticProvider(servermode.DegradedMode)
+	var modeProvider cluster.ChangeEventProvider
```

Comment:
does it make sense to extract this "logic" into a common modeProvider producer method, or we don't forsee that this will be used anywhere else?